### PR TITLE
bugfix: verify presence of the given version

### PR DESCRIPTION
**Pull request by Totktonada**
The action installs tarantool using the following command.

```sh
sudo apt-get install -y tarantool=${version}* tarantool-dev=${version}*
```

If there is no exactly matching version, `apt-get` installs an available one.

For example, Ubuntu Jammy (22.04) has tarantool-2.6.0 in distro's repository and an attempt to install a non-existing tarantool version gives tarantool-2.6.0.

This behavior makes things especially confusing when the `ubuntu-latest` [runner label][1] changes its meaning. For example, when `ubuntu-latest` meant Ubuntu Focal (20.04) the action successfully installs tarantool-1.10.6. Later, `ubuntu-latest` was updated to Ubuntu Jammy (22.04) and the same action invocation installs 2.6.0, because there is no 1.10.6 package for Ubuntu Jammy.

After this patchset such an update doesn't lead to installation of an unexpected tarantool version: it leads to an explicit error.

The patchset also eliminates the 'unable to locate executable file' error after failed attempt to install tarantool. It makes the error diagnostics more clear.

Part of #36

[1]: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#choosing-github-hosted-runners

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -363,23 +369,28 @@ async function run_linux(): Promise<void> {
...
-    await run_linux()
+    success = await run_linux()
   } else {
     core.setFailed(`Action doesn't support ${process.platform} platform`)
   }
 
-  await exec.exec('tarantool --version')
+  if (success) {
+    await exec.exec('tarantool --version')
+  }
```
Comment (grafin) [inline src/main.ts:393]: Wouldn't it be clearer, if we enclose the `run_linux()` call in a `try-catch` block and just move the `tarantool --version` call into that block?

**Review by grafin — COMMENTED** [review]
Thank you for the patches, only one comment from me

File: src/main.ts (hunk 1 above)
Comment (Totktonada) [reply src/main.ts:393]: The `run_linux()` function already has a try-catch block inside. I can move it outside.

```diff
diff --git a/src/main.ts b/src/main.ts
index 0526d6d..8c10824 100644
--- a/src/main.ts
+++ b/src/main.ts
@@ -268,7 +268,6 @@ function dpkg_is_file_included(
 }
 
 async function run_linux(): Promise<void> {
-  try {
   const distro = await lsb_release()
   const distro_id = (await lsb_release_id()).toLowerCase()
   const cache_dir = 'cache-tarantool'
@@ -369,23 +368,25 @@ async function run_linux(): Promise<void> {
   }
 
   await io.rmRF(cache_dir)
-  } catch (error: unknown) {
-    if (error instanceof Error) {
-      core.setFailed(error.message)
-    } else {
-      core.setFailed(`No error message`)
-    }
-  }
 }
 
 export async function run(): Promise<void> {
-  if (process.platform === 'linux') {
-    await run_linux()
-  } else {
+  if (process.platform !== 'linux') {
     core.setFailed(`Action doesn't support ${process.platform} platform`)
+    return
   }
 
-  await exec.exec('tarantool --version')
+  await run_linux()
+    .then(_ => {
+      return exec.exec('tarantool --version')
+    })
+    .catch(error => {
+      if (error instanceof Error) {
+        core.setFailed(error.message)
+      } else {
+        core.setFailed(`No error message`)
+      }
+    })
 }
 
 // Export core.setOutput() to use in testing of setup-tarantool.
```

(The diff is created with `-b` to skip indent changes.)

Does it look better?

Comment (grafin) [reply src/main.ts:393]: Yes! Now the `run_linux()` call and it's error handling is done in one place. And the platform compatibility check is separate.

Comment (Totktonada) [reply src/main.ts]: OK, I've updated it this way.

**Review by grafin — APPROVED** [review]

**Pull request merged**